pkg/tunnelengine/tunnelagent.go

```diff
@@ -86,12 +86,20 @@ func (c *TunnelHandler) Handler() error {
 	for i := range gws.Items {
 		// try to update public IP if empty.
 		gw := &gws.Items[i]
-		if ep := getTunnelActiveEndpoints(gw); ep != nil && ep.PublicIP == "" {
-			err := c.configGatewayPublicIP(gw)
-			if err != nil {
-				klog.ErrorS(err, "error config gateway public ip", "gateway", klog.KObj(gw))
+		if ep := getTunnelActiveEndpoints(gw); ep != nil {
+			if ep.PublicIP == "" || ep.NATType == "" || ep.PublicPort == 0 && ep.NATType != utils.NATSymmetric {
```

Comment:
It can be deleted?

Reply:
I'm sorry I didn't understand? All I've done here is add the logic to call `configGatewayStunInfo`, and not remove the logic to call `configGatewayPublicIP`

Reply:
L-90 can be deleted

Reply:
Since we need to execute `continue` at the end of this if, is your suggestion for me to remove this if and move the `continue` logic into two of the if logic? Just like this?:
```go
if ep.PublicIP == "" {
	if err := c.configGatewayPublicIP(gw); err != nil {
		klog.ErrorS(err, "error config gateway public ip", "gateway", klog.KObj(gw))
	}
	continue
}
if ep.NATType == "" || ep.PublicPort == 0 && ep.NATType != utils.NATSymmetric {
	if err := c.configGatewayStunInfo(gw); err != nil {
		klog.ErrorS(err, "error config gateway stun info", "gateway", klog.KObj(gw))
	}
	continue
}
```

Reply:
According to my understanding, it can be achieved as follows:

Reply:
`
if ep := getTunnelActiveEndpoints(gw); ep != nil {
	if ep.PublicIP == "" {
		if err := c.configGatewayPublicIP(gw); err != nil {
			klog.ErrorS(err, "error config gateway public ip", "gateway", klog.KObj(gw))
		}
	}
	if ep.NATType == "" || ep.PublicPort == 0 && ep.NATType != utils.NATSymmetric {
		if err := c.configGatewayStunInfo(gw); err != nil {
			klog.ErrorS(err, "error config gateway stun info", "gateway", klog.KObj(gw))
		}
	}
	continue
}
`

Reply:
Keep it the way it was